locals.tf

```diff
@@ -9,9 +9,15 @@ locals {
   }
   default_tier = var.environment == "production" ? "db-custom-2-8192" : "db-f1-micro"
 
+  db_engine = split("_", var.database_version)[0]
+
   backup_config = defaults(var.backup_config, local.default_backup_config)
   labels        = merge(local.default_labels, var.labels)
   storage_size  = var.storage_autoresize == true ? null : var.storage_size
   tier          = var.tier != null ? var.tier : local.default_tier
-  users         = { for user in var.users : "${user.name}@${user.host}" => user }
+
+  mysql_users = local.db_engine != "MYSQL" ? {} : {
+    for user in var.users : "${user.name}@${user.host == null ? "%" : user.host}" => defaults(user, { host = "%" })
```

Comment:
Having a bit of trouble making sense of this line. Could use some documentation perhaps?

Reply:
D'oh, somehow I totally misread/misinterpreted the usage of `defaults()` here :facepalm: